lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,421 @@
+        self.assertArrayAlmostEqual(result[1].data.mask, expected_mask_t1)
+
+    def test_returns_expected_values_10_minutes(self):
+        """Test function returns the expected accumulations over the complete
+        10 minute aggregation period. These are written out long hand to make
+        the comparison easy. Note that the test have been constructed such that
+        only the top row is expected to show a difference by including the last
+        5 minutes of the accumulation, all the other results are the same as
+        for the 5 minute test above."""
+
+        expected_t0 = np.array([
+            [0.015, 0.045, 0.075, 0.105, 0.135,
+             0.195, 0.285, 0.375, 0.465, 0.555],
+            [0.015, 0.045, 0.075, 0.105, 0.135,
+             np.inf, np.inf, np.inf, np.inf, np.inf],
+            [0., 0., 0., 0., 0.,
+             np.inf, np.inf, np.inf, np.inf, np.inf],
+            [0., 0., 0., 0., 0.,
+             0.045, 0.135, 0.225, 0.315, 0.405]])
+
+        expected_mask_t0 = np.array([
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+
+        plugin = Accumulation(accumulation_period=600, accumulation_units='mm')
+        result = plugin.process(self.cubes)
+
+        self.assertArrayAlmostEqual(result[0].data, expected_t0)
+        self.assertArrayAlmostEqual(result[0].data.mask, expected_mask_t0)
+
+    def test_returns_total_accumulation_if_no_period_specified(self):
+        """Test function returns a list containing a single accumulation cube
+        that is the accumulation over the whole period specified by the rates
+        cubes. The results are the same as the 10 minute test above as that is
+        the total span of the input rates cubes."""
+
+        expected_t0 = np.array([
+            [0.015, 0.045, 0.075, 0.105, 0.135,
+             0.195, 0.285, 0.375, 0.465, 0.555],
+            [0.015, 0.045, 0.075, 0.105, 0.135,
+             np.inf, np.inf, np.inf, np.inf, np.inf],
+            [0., 0., 0., 0., 0.,
+             np.inf, np.inf, np.inf, np.inf, np.inf],
+            [0., 0., 0., 0., 0.,
+             0.045, 0.135, 0.225, 0.315, 0.405]])
+
+        expected_mask_t0 = np.array([
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+
+        plugin = Accumulation(accumulation_units='mm')
+        result = plugin.process(self.cubes)
+
+        self.assertArrayAlmostEqual(result[0].data, expected_t0)
+        self.assertArrayAlmostEqual(result[0].data.mask, expected_mask_t0)
+
+    def test_returns_expected_values_1_minute(self):
+        """Test function returns the expected accumulations over a 1 minute
+        aggregation period. In this case there is no aggregation, so the input
+        cube should be returned as the output cube."""
```

Comment:
Not any more I think, now we are summing the mid-points (so the first result should be 0.5*(cube0+cube1) etc).

Would be good to move the length test here - see comment on test above.

Reply:
Test moved into here and doc-string trimmed.